diff --git a/erts/emulator/beam/packet_parser.c b/erts/emulator/beam/packet_parser.c
index f14910bc72..20c9ce622c 100644
--- a/erts/emulator/beam/packet_parser.c
+++ b/erts/emulator/beam/packet_parser.c
@@ -797,18 +797,6 @@ int packet_parse_http(const char* buf, int len, int* statep,
         while (!is_tspecial((unsigned char)*ptr)) {
             if (name_len < HTTP_MAX_NAME_LEN) {
                 int c = *ptr;
-                if (up) {
-                    if (islower(c)) {
-                        c = toupper(c);
-                    }
-                    up = 0;
-                }
-                else {
-                    if (isupper(c))
-                        c = tolower(c);
-                    else if (c == '-')
-                        up = 1;
-                }                            
                 name_buf[name_len] = c;
                 hash_update(h, c);
             }
